### PR TITLE
Handle STS Credentials Region

### PR DESCRIPTION
**Pull request by ben-taussig-solo**
# Description
 - Partially resolves an issue where assuming a secondary role using an EKS ServiceAccount would fail when the region of the EKS cluster/OIDC provider used to authenticate the assumeRole request was not us-east-1
   - Prior to this PR, all STS requests were directed to sts.us-east-1.amazonaws.com
   - When a user has configured IAM Roles for EKS ServiceAccounts, the OIDC provider associated with that cluster will be scoped to the same region that the cluster is in. Subsequent IAM requests made after assuming credentials from that provider must be sent to the AWS STS endpoint that matches the region of the cluster, i.e. sts.us-east-2.amazonaws.com for a cluster/OIDC provider in us-east-2
   - For this reason, users with EKS clusters outside of us-east-1 were unable to use the [Gloo Edge IRSA role-chaining workflow](https://docs.solo.io/gloo-edge/latest/guides/traffic_management/destination_types/aws_lambda/cross-account/) before these changes after aws shored up its cross region access rights
 - The Gloo OSS PR [here](https://github.com/solo-io/gloo/pull/8605/) provides helm changes that, along with the changes in this PR resolve this issue

### CODE REVIEW

File: api/envoy/config/filter/http/aws_lambda/v2/aws_lambda.proto [hunk 1]
```diff
@@ -101,6 +101,8 @@ message AWSLambdaConfig {
     string uri = 2 [ (validate.rules).string.min_bytes = 1 ];
     // timeout for the request
     google.protobuf.Duration timeout = 3;
+    // Region for the sts endpoint
+    string region = 4 [ (validate.rules).string.min_bytes = 1 ];
```
[the remark this answers is not in the document]
Comment (nfuden) [reply api/envoy/config/filter/http/aws_lambda/v2/aws_lambda.proto:105]: Should this be required? I think it definitely shouldnt be in the user facing api but a control plane can do the defaults for this message if we need

[the remark this answers is not in the document]
Comment (ben-taussig-solo) [reply api/envoy/config/filter/http/aws_lambda/v2/aws_lambda.proto]: I made it not required [here](https://github.com/solo-io/envoy-gloo/pull/267/commits/ca0f38a029a1f128fb417cfb6b5a0399d4558255) -- the control plane should always set this field on any new serviceAccountCredentials, but we also handle the case where it isn't set in the data plane. I think this is probably the safest way to go about things

**Comment by ben-taussig-solo** [comment]
I made a change here: https://github.com/solo-io/envoy-gloo/pull/267/commits/e11ac28a3271406b5abc4393738504cf4e687dd5 which handles the error seen in the issue when we receive a response from AWS, with testing here: [c48ddb6](https://github.com/solo-io/envoy-gloo/pull/267/commits/c48ddb6eefb9170e20237f515c61d074aab18035)

looking for input as to whether I should add an explicit log message in this case

File: api/envoy/config/filter/http/aws_lambda/v2/aws_lambda.proto [hunk 2]
```diff
@@ -101,6 +101,8 @@ message AWSLambdaConfig {
     string uri = 2 [ (validate.rules).string.min_bytes = 1 ];
     // timeout for the request
     google.protobuf.Duration timeout = 3;
+    // Region for the sts endpoint, defaults to us-east-1
```
Comment (nfuden) [inline api/envoy/config/filter/http/aws_lambda/v2/aws_lambda.proto:104]:
```suggestion
    // Region for the sts endpoint, defaults to us-east-1. 
    // This must be an enabled region https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp_enable-regions.html
    // If the caller lives in an aws region they should use the same endpoint.
```

Comment (ben-taussig-solo) [reply api/envoy/config/filter/http/aws_lambda/v2/aws_lambda.proto]: Updated here, let me know what you think: https://github.com/solo-io/envoy-gloo/pull/267/commits/571ad3930983ba28214295adc54467a8396cccb2

```cpp
    // Region for the sts endpoint, defaults to us-east-1. 
    // This must be an enabled region https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp_enable-regions.html
    // This should match the region specified in the uri.
```

**Review by nfuden — APPROVED** [review]

**Pull request merged**